lib/matplotlib/animation.pyi

```diff
@@ -152,11 +152,17 @@ class HTMLWriter(FileMovieWriter):
     def grab_frame(self, **savefig_kwargs): ...
     def finish(self) -> None: ...
 
+class EventSourceProtocol(Protocol):
+    def add_callback(self, func: Callable): ...
+    def remove_callback(self, func: Callable): ...
+    def start(self): ...
+    def stop(self): ...
+
 class Animation:
     frame_seq: Iterable[Artist]
-    event_source: Any
+    event_source: EventSourceProtocol
```

Comment:
`Animation._stop` sets this attribute to `None`, so this annotation should still have `| None`, though I don't know if that could be replaced by a plain call to `.stop`.

Reply:
Changed to 
```
event_source: EventSourceProtocol | None  # TODO: We should remove None
```

It's a bit weird that we set this to None. It's a a core property of Animation and setting to None means that the Animation is not usable anymore after `_stop()`. This looks like early cleanup. Not sure whether that is needed.

It came in via https://github.com/matplotlib/matplotlib/pull/22561. @greglucas Was there a specific reason to remove the event source.

Reply:
I think this was to remove continuous timers. I don't recall why I didn't call "stop" on the event_source instead of removing it entirely. I wonder if there was a test / external event source that didn't implement "stop" 🤷 

I think your suggestion makes sense and it was likely just overlooked previously with no specific need to clean it up.